sunpy/net/dataretriever/sources/ace.py

```diff
@@ -0,0 +1,336 @@
+    2016-05-20 00:00:00 2016-05-21 00:00:00    ACE     epam]
+    
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+
+        START_DATE = datetime.datetime(2015, 7, 29)
+        base_url = 'ftp://ftp.swpc.noaa.gov/pub/lists/ace/'
+        if timerange.start < START_DATE:
+            raise ValueError("The earliest date for which EPAM data is available is {:%Y-%m-%d}".format(START_DATE))
+        total_days = (timerange.end - timerange.start).days + 1
+        all_days = timerange.split(total_days)
+        result = list()
+        for day in all_days:
+            url = base_url + "{date:%Y%m%d}_ace_epam_5m.txt".format(date = day.end)
+            result.append(url)
+
+        if ((datetime.datetime.now() - timerange.end).days == 0):
+            url = base_url + 'ace_epam_5m.txt'
+            result.append(url)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['source'] = 'ACE'
+        self.map_['instrument'] = 'epam'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+        
+        Parameters
+        ----------
+        query : list of query objects
+        
+        Returns
+        -------
+        boolean: answer as to whether client can service the query
+        
+        """
+        chkattr = ['Time', 'Instrument']
+        chklist = [x.__class__.__name__ in chkattr for x in query]
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'epam':
+                return all(chklist)
+        return False
+
+
+class MAGClient(GenericClient):
+     """
+    Returns a list of URLS to ACE SWEPAM files corresponding to value of input timerange.
+    URL source: `ftp://ftp.swpc.noaa.gov/pub/lists/ace/`.
+    The earliest date available is from 29-Jul-2015
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'mag'
+    
+    Returns
+    -------
+    urls: list
+          list of urls corresponding to requested time range.
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(a.Time('2015/12/28 00:00:00', '2015/12/28 00:03:00'), a.Instrument('mag'))
+    >>> print(results)
+    >>> [<Table length=3>
+         Start Time           End Time      Source Instrument
+            str19               str19         str3     str6   
+    ------------------- ------------------- ------ ----------
+    2016-05-18 00:00:00 2016-05-19 00:00:00    ACE     mag
+    2016-05-19 00:00:00 2016-05-20 00:00:00    ACE     mag
+    2016-05-20 00:00:00 2016-05-21 00:00:00    ACE     mag]
+    
```

Comment:
is that a copy paste?
One of the tests runs these docstrings... so if you get them wrong the tests fails.